main/src/com/google/refine/util/LangDetectUtils.java

```diff
@@ -0,0 +1,38 @@
+package com.google.refine.util;
+
+
+import com.google.common.base.Optional;
+import com.optimaize.langdetect.LanguageDetector;
+import com.optimaize.langdetect.LanguageDetectorBuilder;
+import com.optimaize.langdetect.i18n.LdLocale;
+import com.optimaize.langdetect.ngram.NgramExtractors;
+import com.optimaize.langdetect.profiles.LanguageProfile;
+import com.optimaize.langdetect.profiles.LanguageProfileReader;
+import com.optimaize.langdetect.text.CommonTextObjectFactories;
+import com.optimaize.langdetect.text.TextObject;
+import com.optimaize.langdetect.text.TextObjectFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+public class LangDetectUtils {
+
+    public static Optional<LdLocale> detect(String text) throws IOException {
+        List<LanguageProfile> languageProfiles = new LanguageProfileReader().readAllBuiltIn();
```

Comment:
Here you are loading all the languages, which I expect can take a bit of time. Since the language detection function will typically be called many times (for instance once per row in the dataset), it would be good to do that just once, the first time the function is called.

Reply:
I see. Thanks for this!